hoodie-client/src/main/java/com/uber/hoodie/index/bloom/IndexFileRangeNode.java

```diff
@@ -0,0 +1,142 @@
+/*
+ *  Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.uber.hoodie.index.bloom;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a node in the {@link IndexLookUpTree}. Holds information pertaining to a single index file, viz file name
+ * min record key and max record key.
+ */
+class IndexFileRangeNode implements Comparable<IndexFileRangeNode>, Serializable {
+
+  private final List<String> fileNameList = new ArrayList<>();
+  private final String minRecordKey;
+  private final String maxRecordKey;
+  private String rightSubTreeMax = null;
```

Comment:
since you work with Comparable anyway to do the matching? Should all of these strings just be `Comparable` objects? Code remains pretty generic then? wdyt>?

Reply:
having minRecordKey and maxRecordKey be of type `Comparable`

Reply:
This is doable. but punting for now